pkg/planner/util/path.go

```diff
@@ -383,31 +383,40 @@ func (c1 Col2Len) dominate(c2 Col2Len) bool {
 	return true
 }
 
-// CompareCol2Len will compare the two Col2Len maps. The last return value is used to indicate whether they are comparable.
-// When the second return value is true, the first return value:
+// CompareCol2Len will compare the two Col2Len maps.
+// The first return value:
 // (1) -1 means that c1 is worse than c2;
 // (2) 0 means that c1 equals to c2;
 // (3) 1 means that c1 is better than c2;
+// The 2nd return value is used to indicate whether they are comparable. If they are NOT comparable, then the caller
+// should use other criteria to determine whether the winner is justified.
 func CompareCol2Len(c1, c2 Col2Len) (int, bool) {
 	l1, l2 := len(c1), len(c2)
 	if l1 > l2 {
 		if c1.dominate(c2) {
 			return 1, true
 		}
-		return 0, false
+		return 1, false
```

Comment:
Just double check, I searched some caller of this function. Seem when return `..., false` there's no usage of the first return value. And in this PR the first return value is more precise. Do you plan to add usage on it in `..., false` case?